test/integration/puppetlabs/services/jruby_pool_manager/jruby_pool_int_test.clj

```diff
@@ -156,9 +156,8 @@
        (is (= 4 (.size (jruby-core/get-pool pool-context))))
        (jruby-core/flush-pool-for-shutdown! pool-context)
 
-       ;; flushing the pool removes all JRubyInstances but causes ShutdownPoisonPills
-       ;; to be added, so it should still have 4 instances.
-       (is (= 4 (.size pool)))
+       ;; flushing the pool should remove all JRubyInstances
+       (is (= 0 (.size pool)))
        ;; But they should all be shutdown poison pills
        (is (every? jruby-schemas/shutdown-poison-pill? (.getRegisteredElements pool)))))))
```

Comment:
Should this still be true with the poison pill rework?  Wasn't clear to me that a pill would end up being a registered element in the pool.